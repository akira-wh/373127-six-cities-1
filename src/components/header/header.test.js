import React from 'react';
import renderer from 'react-test-renderer';

import Header from './header.jsx';

describe(`Отрисовка представления Header`, () => {

  it(`Главная страница --> Ссылка логотипа неактивна.`, () => {
    const userLogin = `markwatney@arescrew.com`;
    const isRenderedOnMainPage = true;

    const view = renderer.create(
        <Header
          isRenderedOnMainPage={isRenderedOnMainPage}
          userLogin={userLogin}
        />
    ).toJSON();

    expect(view).toMatchSnapshot();
  });

  it(`Второстепенные страницы --> Ссылка логотипа активна.`, () => {
    const userLogin = `markwatney@arescrew.com`;

    const view = renderer.create(<Header userLogin={userLogin} />).toJSON();

    expect(view).toMatchSnapshot();
  });

  it(`Пользователь авторизирован (логин - непустая строка) --> Отображается логин.`, () => {
    const userLogin = `markwatney@arescrew.com`;

    const view = renderer.create(<Header userLogin={userLogin} />).toJSON();

    expect(view).toMatchSnapshot();
  });

  it(`Пользователь неавторизирован (логин - пустая строка) --> Отображается предложение авторизации.`, () => {
    const userLogin = ``;

    const view = renderer.create(<Header userLogin={userLogin} />).toJSON();

    expect(view).toMatchSnapshot();
  });
});
