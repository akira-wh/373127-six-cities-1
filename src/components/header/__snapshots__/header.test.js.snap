// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Отрисовка представления Header Второстепенные страницы --> Ссылка логотипа активна. 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="main.html"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item  user"
          >
            <a
              className="header__nav-link  header__nav-link--profile"
              href="#"
            >
              <div
                className="header__avatar-wrapper  user__avatar-wrapper"
              />
              <span
                className="header__user-name  user__name"
              >
                markwatney@arescrew.com
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Отрисовка представления Header Главная страница --> Ссылка логотипа неактивна. 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link  header__logo-link--active"
          href={null}
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item  user"
          >
            <a
              className="header__nav-link  header__nav-link--profile"
              href="#"
            >
              <div
                className="header__avatar-wrapper  user__avatar-wrapper"
              />
              <span
                className="header__user-name  user__name"
              >
                markwatney@arescrew.com
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Отрисовка представления Header Пользователь авторизирован (логин — непустая строка) --> Отображается логин. 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="main.html"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item  user"
          >
            <a
              className="header__nav-link  header__nav-link--profile"
              href="#"
            >
              <div
                className="header__avatar-wrapper  user__avatar-wrapper"
              />
              <span
                className="header__user-name  user__name"
              >
                markwatney@arescrew.com
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;

exports[`Отрисовка представления Header Пользователь неавторизирован (логин — пустая строка) --> Отображается предложение авторизации. 1`] = `
<header
  className="header"
>
  <div
    className="container"
  >
    <div
      className="header__wrapper"
    >
      <div
        className="header__left"
      >
        <a
          className="header__logo-link"
          href="main.html"
        >
          <img
            alt="6 cities logo"
            className="header__logo"
            height="41"
            src="img/logo.svg"
            width="81"
          />
        </a>
      </div>
      <nav
        className="header__nav"
      >
        <ul
          className="header__nav-list"
        >
          <li
            className="header__nav-item  user"
          >
            <a
              className="header__nav-link  header__nav-link--profile"
              href="#"
            >
              <div
                className="header__avatar-wrapper  user__avatar-wrapper"
              />
              <span
                className="header__login"
              >
                Sign in
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
